File: Arrays and Strings/is_unique.cpp
#include <iostream>
#include <set>

bool is_unique(const std::string& str)
{
    std::set<char> occurred;
    for(int i = 0; i < str.size(); i++)
    {
        if(occurred.find(str[i]) == occurred.end())
            occurred.insert(str[i]);
        else
            return false;
    }
    return true;
}

int main()
{
    std::cout << is_unique("hel") << std::endl;
    std::cout << is_unique("hell") << std::endl;
    std::cout << is_unique("helol") << std::endl;
    std::cout << is_unique("") << std::endl;
}

File: Arrays and Strings/is_unique.py
def is_unique(s) -> bool:
    dict = {}
    for letter in s:
        if letter in dict:
            return False
        else:
            dict[letter] = True
    return True

print(is_unique("hel"))
print(is_unique("hell"))
print(is_unique("helol"))
print(is_unique(""))